metrics/resource_view_test.go

```diff
@@ -189,6 +196,69 @@ func TestAllMetersExpiration(t *testing.T) {
 	// (123=9m, 456=evicted, 789=0m)
 }
 
+func TestIfAllMeterResourcesAreRemoved(t *testing.T) {
+	allMeters.clock = clock.Clock(clock.NewFakeClock(time.Now()))
+	var fakeClock *clock.FakeClock = allMeters.clock.(*clock.FakeClock)
+	ClearMetersForTest() // t+0m
+	// Register many resources at once
+	for i := 1; i <= 100; i++ {
+		res := resource.Resource{Labels: map[string]string{"foo": "bar"}}
+		res.Labels["id"] = fmt.Sprint(i)
+		if _, err := optionForResource(&res); err != nil {
+			t.Error("Should succeed getting option, instead got error ", err)
+		}
+		m := stats.Int64("testView_sum", "", stats.UnitDimensionless)
+		v := view.View{Name: fmt.Sprintf("testview-%d", i), Measure: m, Aggregation: view.Sum()}
+
+		err := RegisterResourceView(&v)
+		if err != nil {
+			t.Fatal("RegisterResourceView =", err)
+		}
+	}
+	allMeters.lock.Lock()
+	// Make a copy to test against as allMeters should be cleaned up
+	copyAllMeters := make(map[string]*meterExporter)
+	for k, v := range allMeters.meters {
+		copyAllMeters[k] = v
+	}
+	allMeters.lock.Unlock()
+	resourceViews.lock.Lock()
+	for _, meter := range copyAllMeters {
+		for _, mView := range resourceViews.views {
+			want, got := mView, meter.m.Find(mView.Name)
+			if got == nil {
+				t.Errorf("View %s is not available", want.Name)
+			} else if want.Name != got.Name {
+				t.Errorf("Want %v, got %v", want.Name, got.Name)
+			}
+		}
+	}
+	resourceViews.lock.Unlock()
```

Comment:
wonder if it's worth reworking this a bit so that we can defer the Unlocks

Reply:
Will take a look.

Reply:
If copying is done in a separate func then yes but at after copying we need to progress the clock which means we have to unlock to allow the cleanup goroutine to run, we can simply defer the Unlocks for end of the test.

Reply:
you can just put it into an anonymous
```
func() {
lock
defer unlock
do stuf
}()
```